Calliope Mini 3

HaraldRau opened the pull request:
## Bauteil nach Revision und Fehler beim Importieren ##

- fzpz-Datei aus dem Forum extrahiert und umbenannt
- breadboard-svg im Zweig gelöscht und ersetzt durch neue Breadbord-svg

### Quelle: ###
[https://forum.fritzing.org/t/calliope-3-als-bauteil-nach-fritzing/24107/23](https://forum.fritzing.org/t/calliope-3-als-bauteil-nach-fritzing/24107/23)

[Calliope_mini_3-connectors-fixed.fzpz (50.3 KB)](https://forum.fritzing.org/uploads/short-url/hvyU95Uwe3pwpHjxKMD53yvRrO2.fzpz)

[Calliope_mini_3_mit_LED](https://forum.fritzing.org/uploads/short-url/4sOvf6qEVpfyMSW0UAyksJAkXio.fzpz)

HaraldRau commented:
Good morning,

Calliope Mini 3 from the Forum

[https://forum.fritzing.org/t/calliope-3-als-bauteil-nach-fritzing/24107/34](https://forum.fritzing.org/t/calliope-3-als-bauteil-nach-fritzing/24107/34)

Now it worked. Thanks to Peter's work, no further errors occurred.

Thank you all for your help.

KjellMorgenstern commented:
Connectors on the left Jacdac are labeled jadac1-*
Connectors on the right Jacdac are a mix of jadac1-* and jadac2-*

Is this intended, or maybe label them jacdac1-* and jacdac2-* ?

HaraldRau commented:
Yes, I think that's a mistake. The correct name should be:
jadac1-1/D
jadac1-2/GND
jadac1-3/3V3

jadac2-1/D
jadac2-2/GND
jadac3-3/3V3

Thank you for troubleshooting!

HaraldRau commented:
I will change it in the fzp-file.

HaraldRau committed:
Update calliope-mini-3_2.fzp

Change
<description>jadac1-2/GND</description>
to
<description>jadac2-2/GND</description>
and
<description>jadac1-3/3V3</description>
to
<description>jadac2-3/3V3</description>

HaraldRau commented:
I found some disconnected connections. They weren't labeled on the board. I2C and UART are routed to the PIN header.
I checked it with the voltmeter.
 I'll change that tonight!

HaraldRau committed:
Update calliope-mini-3_2.fzp

Einfügen von
<bus id="SCL">
<bus id="SDA">
<bus id="RX">
<bus id="TX">

HaraldRau commented:
I found another error. On the breadboard, pin 23 (GND) is connected to JP1. In the PCB view, this connection is not routed and shows a connection from JP1 to pin 19. I am looking for the error!
![BME_280_Calliope_Steckplatine](https://github.com/fritzing/fritzing-parts/assets/47145786/fc90bbd3-9a35-436b-ab7f-1cb8af663b61)
![BME_280_Calliope_Leiterplatte](https://github.com/fritzing/fritzing-parts/assets/47145786/da8108e8-d68e-46a5-b2bc-fe323c68b2f1)

HaraldRau commented:
Is it better to close the pull request for now?

KjellMorgenstern commented:
No need to close it, just push fixes to it as they come in. 

Besides, I am not sure if this is an error. Pin 19, 22 and 23 all are GND. 
Fritzing will always select the closest connector of the subnet to draw the ratsnest line.

HaraldRau commented:
Ah, that's right. On the old Calliope, pin 19 was I2C; now it is GND. I will still check it again!
Thanks!

HaraldRau commented:
I just panicked a bit; everything is fine!
![BME_280_Calliope_Leiterplatte](https://github.com/fritzing/fritzing-parts/assets/47145786/29edc634-2899-448a-be73-f1ddde00850e)
![BME_280_Calliope_Steckplatine](https://github.com/fritzing/fritzing-parts/assets/47145786/56fa94fe-671c-41a1-8f23-d3c066066c24)

HaraldRau committed:
Update calliope-mini-3_2_1_breadboard.svg

A member of the Calliope team found an error in the visual representation. The LEDs were not displayed, so I adjusted that. The fzpz file worked well and was tested in the forum. Now I’ve tried to make the change in the pull request. Unfortunately, my Inkscape generates different XML code. Each object has a "style" entry. Could this be a problem?

HaraldRau commented:
I think the XML code is not correct. In the previous version, the "style" element was defined at the beginning of the document. The current version has a "style" element for each object again.

Are there any invalid entries in the "style" elements?

` style="fill:none;fill-opacity:0.712946;stroke:#faf1a9;stroke-width:0.773404;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1;paint-order:stroke markers fill" `

KjellMorgenstern commented:
When exporting (from Inkscape) could you try "optimized", and check the SVG 1.1 box?

This will apply "scour" to optimize the SVG. It avoid conflicts between styles and attributes, possibly prevent the SVG from containing styles that Fritzing does not support. 

![image](https://github.com/user-attachments/assets/4f4ff35b-62a1-4ab4-a409-b84a5fab93a1)

HaraldRau commented:
I did that, the source code looks much cleaner to me now. Thank you very much for the tip. I'll take another look at the file in Git and review the changes.
Gruß Harald!

HaraldRau committed:
Update calliope-mini-3_2_1_breadboard.svg

Export from Inkscape,  "optimized" and SVG 1.1

HaraldRau committed:
Update calliope-mini-3_2_1_icon.svg

Add LED on icon from inkscape optimized SVG 1.1

vanepp commented:
@KjellMorgenstern  Where are the settings you referred to in Inkscape? I can't find them in preferences (on Inkscape 1.3.2) are they somewhere else?

HaraldRau commented:
The settings must be activated when saving. If you select the "Optimized SVG" settings, an optimization script is started. You can then select a few options in a dialog.
**File/save Dialog**
![Bildschirmfoto vom 2024-09-04 17-59-57](https://github.com/user-attachments/assets/c399df03-d72f-490a-b4f0-dbf7b9acc4f6)

**Optimierungsdialog**
![Bildschirmfoto vom 2024-09-04 18-00-35](https://github.com/user-attachments/assets/30e0501c-dfac-47e4-8b92-05eab8e5b732)

vanepp commented:
Ah! That works a little better, but I still don't see the dialog that allows selecting export as svg 1.1 anywhere. I have been using plain svg (which was the original recommendation from years ago) like this 

![capture](https://github.com/user-attachments/assets/4b450e0e-d75a-431b-99fc-23251f7bd093)

changing to optimized svg does indeed bring up more dialogs but not the ones I want

![capture1](https://github.com/user-attachments/assets/9b151dfb-808c-430e-9bd4-d3e7dbe03208)

![capture2](https://github.com/user-attachments/assets/07268d60-1999-49a1-9497-09eed8ebe6a2)

![capture3](https://github.com/user-attachments/assets/7ea509eb-0ce5-421d-8a64-4727f56e9f95)

nothing that sets the 1.1 output.

HaraldRau commented:
I'm using Inkscape 1.2.2 on Ubuntu 24.04.1. I think that's the reason for the missing option.

KjellMorgenstern commented:
When selecting "Remove unused IDs" , I find it useful to Preserve IDs starting with
"conn,bread,copper,silk,pcb,schematic,icon"

That list might not be complete. Not all IDs important to Fritzing end with digits.

mMerlin commented:
I have only "conn" in Preserve IDs starting with. For Preserve the following IDs, I have "icon,breadboard,schematic,copper0,copper1". Tweaked over time.

Scour is a python app that can be run standalone. See https://github.com/scour-project/scour. Apache-2.0 license. I'm pretty sure that anything that can selected from the Inkscape dialog can be done from the command line (or a script) as well. A possible tool to add to the CI.

@vanepp The svg 1.1 setting is (for me, Inkscape 1.3.2) on the Save As dialog. You can see it bottom left of the first image in https://github.com/fritzing/fritzing-parts/pull/396#issuecomment-2329458670. Even if it is not in English, the "SVG 1.1" text by the checkbox is enough to point to it. AFAIR, that checkbox has been there back several versions of Inkscape.

KjellMorgenstern commented:
@mMerlin Yes, I apply it all the time. The CI can't modify the files, but maybe this is possible: https://github.com/fritzing/fritzing-parts/issues/397

vanepp commented:
@mMerlin  Apparently only on Linux. On Win10 I get only this:

![capture](https://github.com/user-attachments/assets/88ecfe95-0503-4789-a97f-a762e5a7656d)

no other options that I can find. Clicking on save brings up another menu but it also doesn't have anything to do with the svg level. I'll boot up Mint and load Inkscape and try it there and/or look over the Inkscape forums.

KjellMorgenstern commented:
I tried the "Optimized SVG" on Windows 10, Inkscape 1.3.2. 
The dialog mostly looks like in your screenshots above, there is no option to directly use SVG 1.1.

The screenshot I posted was from Inkscape 1.1.2 that ships with ubuntu 22.

HaraldRau committed:
Merge branch 'fritzing:develop' into develop
